tests/end2end/README.md

```diff
@@ -0,0 +1,28 @@
+# Git-Theta End-2-End tests.
+
+This directory contains a collection of end-2-end tests for git-theta.
+
+## Running the Tests
+
+Each subdirectory represents a test that actually interacts with git. The `runner.sh` script is responsible for running them and reports if they passed for failed. All tests can be run with `./runner.sh`. Additionally, the `.github/workflows/end2endtest.yml` configures these tests to run via GitHub Actions.
+
+## Anatomy of a Test
+
+Each test is a directory and includes a `test.sh` script. Execution of this script runs the test and pass/fail is determined by its exit code (`0` means pass),
+
+Each test also includes a `clean.sh` script. This is run before and after the test to clean up any generated artifacts. This script should never have a non-zero exit code and should be idempotent (we can run it multiple times).
+
+The command `make-test.sh "${test name}"` can be used to create the skeleton of a new test.
+
+### `test.sh`, the (Vegan-)?Meat and Potatoes
+
+The first steps of a `test.sh` file include sourcing the `../utils.sh` so it can use some of our shared functions. Then it should run `set -e` to ensure that errors in part of the test cause the whole test to fail. It should also call `test_init`, which will create a git repo (with a `main` branch) for it and ensure that `set -e` was used.
+
+The next step is often to create and modify some model. The provided `../model.py` file can be used as a helper to create new models and updates with special forms. This script creates two copies of the model, one that lives in the path that is version controlled and one that include information about how it was created. This path is returned on stdout, can be captured in `test.sh`. This checkpoint is not version controlled, uses the native checkpoint format, and should be removed by the clean script.
```

Comment:
- that include*s* information
- *and* can be captured
- What do you mean by the "native" checkpoint format?

Reply:
Fixed

native means deep-learning framework specific, added that